/**
 * Copyright 2021 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package cluster;

service ClusterService {
  rpc RegisterMemberToChief(MemberRequest) returns (ClusterResponse) {}
  rpc QuerryAllNodes(MemberRequest) returns (ClusterResponse) {}
  rpc RegisterFinished(MemberRequest) returns (ClusterResponse) {}
}

enum NodeType {
  kHostNode = 0;
  kDeviceNode = 1;
}

enum ClusterResponseErrorType {
  kClusterSuccess = 0;
  kClusterFailed = 1;
  kClusterWaiting = 2;
}

message NodeInfo {
  NodeType node_type = 1;
  uint32 device_id = 2;
  string ipaddr = 3;
  repeated uint32 rankid = 5;
}

message MemberRequest {
  repeated NodeInfo node_info = 1;
}

message ClusterResponse {
  ClusterResponseErrorType error_code = 1;
  string error_message = 2;
  repeated NodeInfo node_info = 3;
}